windows/devices/uwb/UwbCxAdapterDdiLrp.cxx

```diff
@@ -1635,15 +1623,70 @@ windows::devices::uwb::ddi::lrp::To(const UWB_APP_CONFIG_PARAM &applicationConfi
+                value.insert(::uwb::UwbMacAddress(std::move(data)));
+            }
+        } else { // ::uwb::UwbMacAddressType::Extended
+            for (auto i = 0; i < (applicationConfigurationParameter.paramLength / ::uwb::UwbMacAddress::ExtendedLength); i++) {
+                ::uwb::UwbMacAddress::ExtendedType data{};
+                auto offset = i * ::uwb::UwbMacAddress::ExtendedLength;
+                std::memcpy(std::data(data), &applicationConfigurationParameter.paramValue[0] + offset, std::size(data));
+                value.insert(::uwb::UwbMacAddress(std::move(data)));
+            }
+        }
+
+        uwbApplicationConfigurationParameter.Value = std::move(value);
+    } else {
+        PLOG_ERROR << "unexpected uwb application configuration parameter type encountered, type=" << std::showbase << std::hex << notstd::to_underlying(applicationConfigurationParameter.paramType);
+    }
+
+    return uwbApplicationConfigurationParameter;
+}
+
 std::vector<UwbApplicationConfigurationParameter>
 windows::devices::uwb::ddi::lrp::To(const UWB_APP_CONFIG_PARAMS &applicationConfigurationParameters)
 {
     std::vector<UwbApplicationConfigurationParameter> uwbApplicationConfigurationParameters{};
 
-    detail::ForEachApplicationConfigurationParameter(applicationConfigurationParameters, [&uwbApplicationConfigurationParameters](const auto &applicationConfigurationParameter) {
-        auto uwbApplicationConfigurationParameter = To(applicationConfigurationParameter);
-        uwbApplicationConfigurationParameters.push_back(std::move(uwbApplicationConfigurationParameter));
+    std::optional<::uwb::UwbMacAddressType> macAddressMode;
+    std::optional<UWB_APP_CONFIG_PARAM> destinationMacAddresses;
+    detail::ForEachApplicationConfigurationParameter(applicationConfigurationParameters, [&uwbApplicationConfigurationParameters, &macAddressMode, &destinationMacAddresses](const auto &applicationConfigurationParameter) {
+        // If parameter is MacAddressMode, store value
+        if (To(applicationConfigurationParameter.paramType) == UwbApplicationConfigurationParameterType::MacAddressMode) {
+            macAddressMode = std::get<::uwb::UwbMacAddressType>(To(applicationConfigurationParameter).Value);
+        }
+
+        // If parameter is DestinationMacAddresses, store until other parameters have been converted.
+        if (To(applicationConfigurationParameter.paramType) == UwbApplicationConfigurationParameterType::DestinationMacAddresses) {
+            destinationMacAddresses = applicationConfigurationParameter;
+        } else {
+            auto uwbApplicationConfigurationParameter = To(applicationConfigurationParameter);
+            uwbApplicationConfigurationParameters.push_back(std::move(uwbApplicationConfigurationParameter));
+        }
     });
 
+    // Now that the other parameters have been converted, convert DestinationMacAddresses with special conversion function
+    if (destinationMacAddresses.has_value()) {

```

Comment:
If this is `false`, then there was an error during conversion, right? In that case, should we log an error? I know the nested `To` function will. but we should be printing errors in all the places that are returning bad/incomplete data.